.github/actions/tests/ui-tests-report/action.yaml

```diff
@@ -0,0 +1,32 @@
+name: 'Upload UI Test Report to S3'
+description: 'Uploads Playwright UI test reports to S3 and shares the report link in GitHub step summary'
+author: 'konrad'
+inputs:
+  aws_acces_key_id: 
+    description: 'AWS access key ID'
+    required: true
+  aws_secret_access_key:
+    description: 'AWS secret access key'
+    required: true
+  matrix:
+    description: 'Matrix for the test'
+    required: false  
+runs:
+  using: "composite"
+  steps:
+    - name: Upload Playwright Test Report to S3
+      shell: bash
+      run:
+        npm run upload_report_to_s3
+      env:
+        AWS_ACCESS_KEY_ID: ${{ inputs.aws_acces_key_id }}
```

Comment:
This is a security bad practice. Best to use a role with an explicit permission boundary.
Also, it might be better to consolidate CI assets in one AWS subaccount. Helps at audit time.